### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by vcanales**
To facilitate future improvements to the codebase, this proposes configuring PHPUnit to be able to write tests for the plugin.

This was created using `wp scaffold plugin-test`, plus some modifications to add `wp-cli` and theme data to the test site.

A first test is included at `tests/test-theme-create-theme.php`.

### Testing

To run this suite, a local MySQL installation is required. After that is taken care of, the following commands should get everything setup:

```sh
# install dependencies
composer install
./bin/install-wp-tests.sh
# run tests
npm run test:php
```

### CODE REVIEW

File: admin/class-create-theme.php [hunk 1]
```diff
@@ -30,6 +31,8 @@ public function __construct() {
 		add_action( 'admin_menu', array( $this, 'create_admin_menu' ) );
 		add_action( 'admin_init', array( $this, 'blockbase_save_theme' ) );
 		add_action( 'enqueue_block_editor_assets', array( $this, 'create_block_theme_enqueue' ) );
+
+		$this->theme = wp_get_theme();
```
Comment (mikachan) [inline admin/class-create-theme.php:35]: Does this PR include changes for https://github.com/WordPress/create-block-theme/issues/379? Or are these changes needed for this PR too?

**Review by mikachan — COMMENTED** [review]
Thanks for working on this! It'd be great to have a suite of tests.

The changes are reading well for me. Are there any commands I can try running locally to test this better? Or maybe just ensure that the new test at `tests/test-theme-create-theme.php` passes?

**Comment by vcanales** [comment]
> @mikachan The changes are reading well for me. Are there any commands I can try running locally to test this better? Or maybe just ensure that the new test at tests/test-theme-create-theme.php passes?

I've added testing instructions to the description.

> ### Testing
>
> To run this suite, a local MySQL installation is required. After that is taken care of, the following commands should get everything setup:
>
> ```sh
> # install dependencies
> composer install
> ./bin/install-wp-tests.sh
> # run tests
> npm run test:php
> ```

File: admin/class-create-theme.php (hunk 1 above)
Comment (vcanales) [reply admin/class-create-theme.php:35]: Thanks for catching that! I was using those to verify and they are not needed; I've dropped them from the PR!

Comment (vcanales) [reply admin/class-create-theme.php:35]: @mikachan Added these changes back in order to get checks passing 😅 See, they're already working :)

**Comment by vcanales** [comment]
I added a GitHub actions, which you can see working on this very PR. This required me to add back the changes made for #379 to get tests passing :)

**Comment by pbking** [comment]
Added a command to run tests in watch mode. `npm run test:php:watch`

**Review by pbking — APPROVED** [review]
Ready to test now!
